plotter/db.py

```diff
@@ -188,19 +196,82 @@ class Resource(object):
 
     @classmethod
     def all(cls):
-        return [cls(k).get() for k in cls.db.keys()]
+        return [cls(*tup).get() for tup in cls.db_key_tuples()]
 
     @classmethod
     def db_keys(cls):
         return cls.db.keys()
 
     @classmethod
+    def db_key_tuples(cls):
+        return [k.split(" ") for k in cls.db_keys()]
+
+    def rename(self, **kwargs):
+        """Read data for this key, delete that db entry, and save it under another key"""
+        for k in kwargs:
+            if k not in self.keys:
+                raise ValueError(
+                    "names of arguments must be db keys: " +
+                    ", ".join([str(x) for x in self.keys]))
+        data_dict = self.get()
+        self.put(None)
+        self.info.update(kwargs)
+        self.values = tuple(self.info[k] for k in self.keys)
+        self.set_db_key()
+        self.put(data_dict)
+    
+    @classmethod
+    def rename_all(cls, find, replace):
```

Comment:
`rename_all` is a method for each database class, which calls rename on all entries